### PR TITLE
Add video object detection example

### PR DESCRIPTION
**Pull request by gbryant-arm**
This PR adds a video object detection example.
This example combines and integrates two simpler examples, the video decoder and the deep learning server.
A video decoder based on openh264 decodes an H264 video into individual frames, which are converted to RGB and made palatable to an object detector built on top of the Darknet neural network framework. The output is a list of objects detected associated with their detection probability.

TODO:
* ~~Change README~~
* Try & fine tune other YOLO models (perf isn't epic)
* ~~Fix minor bug in color space transformation: some light pixels saturate when adjacent to darker pixels~~
  * ~~Plan: use darknet's transformation (`stbi__YCbCr_to_RGB_row()`) instead of mine~~
  * ~~NB: this bug doesn't seem to frustrate object detection~~
* ~~Investigate adding a computer vision algo to the pipeline that doesn't process frames that are not substantially different from the previous ones, prior to passing them to the object detection model. That could help speed up the process, as object detection is the bottleneck~~

### CODE REVIEW

File: video-object-detection/.gitignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
+args.cfg
+
+/data/mnist/images/
```
Comment (egrimley-arm) [inline video-object-detection/.gitignore:3]: Absolute path names in a `.gitignore` seems strange to me.

Comment (gbryant-arm) [reply video-object-detection/.gitignore:3]: Yeah. That's a copy of the `.gitignore` in the deep learning server example

File: video-object-detection/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Video object detection example
+
+This example combines and integrates two simpler examples, the video decoder and the [deep learning server](https://github.com/veracruz-project/veracruz-examples/tree/main/deep-learning-server).
+The video decoder leverages [`openh264`](https://github.com/veracruz-project/openh264) to decode an H264 video into individual frames, which are converted to RGB and made palatable to an object detector built on top of the [Darknet neural network framework](https://github.com/mofanv/darknet-src). The output is a list of detected objects, associated with their detection probability, and an optional prediction image showing each detected object in a bounding box.
```
Comment (egrimley-arm) [inline video-object-detection/README.md:4]: Can't we have "uses" instead of "leverages"? :-)

File: video-object-detection/cfg/coco.data [hunk 3]
```diff
@@ -0,0 +1,8 @@
+classes= 80
+train  = /home/pjreddie/data/coco/trainvalno5k.txt
```
Comment (egrimley-arm) [inline video-object-detection/cfg/coco.data:2]: Are these two references to `/home/pjreddie` correct?

File: video-object-detection/README.md (hunk 2 above)
Comment (gbryant-arm) [reply video-object-detection/README.md:4]: Wow, seems to be US slang. Will change it :+1:

File: video-object-detection/cfg/coco.data (hunk 3 above)
Comment (gbryant-arm) [reply video-object-detection/cfg/coco.data:2]: They're not correct. He's the author of Darknet. Will remove them as we don't need them for inference.
FWIW the issue comes from the deep learning server example.

**Comment by gbryant-arm** [comment]
Closing as obsoleted by #9

**Pull request closed without merging**